scaleway/resource_instance_security_group_rules.go

```diff
@@ -0,0 +1,130 @@
+			},
+		},
+	}
+}
+
+func customImporterState(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	// importZID is the ID set by the user when using the Terraform's import function.
+	// It should be a SecurityGroupZID.
+	importZID := d.Id()
+
+	d.Set("security_group_id", importZID)
+	d.SetId(securityGroupRulesZIDFromSecurityGroupZID(importZID))
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceScalewayInstanceSecurityGroupRulesCreate(d *schema.ResourceData, m interface{}) error {
+	d.SetId(securityGroupRulesZIDFromSecurityGroupZID(d.Get("security_group_id").(string)))
+
+	// We call update instead of read as it will take care of creating rules.
+	return resourceScalewayInstanceSecurityGroupRulesUpdate(d, m)
+}
+
+func resourceScalewayInstanceSecurityGroupRulesRead(d *schema.ResourceData, m interface{}) error {
+	securityGroupRulesZID := d.Id()
+	securityGroupZID := securityGroupZIDFromSecurityGroupRulesZID(securityGroupRulesZID)
+
+	instanceApi, zone, securityGroupID, err := instanceAPIWithZoneAndID(m, securityGroupZID)
+	if err != nil {
+		return err
+	}
+
+	d.Set("security_group_id", securityGroupZID)
+
+	inboundRules, outboundRules, err := getSecurityGroupRules(instanceApi, zone, securityGroupID, d)
+	if err != nil {
+		return err
+	}
+
+	d.Set("inbound_rule", inboundRules)
+	d.Set("outbound_rule", outboundRules)
+
+	return nil
+}
+
+func resourceScalewayInstanceSecurityGroupRulesUpdate(d *schema.ResourceData, m interface{}) error {
+	securityGroupZID := securityGroupZIDFromSecurityGroupRulesZID(d.Id())
+	instanceApi, zone, securityGroupID, err := instanceAPIWithZoneAndID(m, securityGroupZID)
+	if err != nil {
+		return err
+	}
+
+	err = updateSecurityGroupeRules(d, zone, securityGroupID, instanceApi)
+	if err != nil {
+		return err
+	}
+
+	return resourceScalewayInstanceSecurityGroupRulesRead(d, m)
+}
+
+func resourceScalewayInstanceSecurityGroupRulesDelete(d *schema.ResourceData, m interface{}) error {
+	securityGroupZID := securityGroupZIDFromSecurityGroupRulesZID(d.Id())
+	instanceApi, zone, securityGroupID, err := instanceAPIWithZoneAndID(m, securityGroupZID)
+	if err != nil {
+		return err
+	}
+
+	d.Set("inbound_rule", nil)
+	d.Set("outbound_rule", nil)
+
+	err = updateSecurityGroupeRules(d, zone, securityGroupID, instanceApi)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// securityGroupRulesZIDFromSecurityGroupZID returns the
+//
+// SecurityGroupRules.ID should be the based on SecurityGroup.ID.
+// This is necessary to support Terraform import feature. -> false -> why is this necessary ?
+// If we want to support multiple SGs(SecurityGroup) for 1 SGRS(SecurityGroupRules),
+// we could always use the first SecurityGroup's ID,
+// because from the API,
+// the data for a single SGRS is duplicated for all SGs using the same SGRS.
+func securityGroupRulesZIDFromSecurityGroupZID(SecurityGroupZID string) string {
+	return securityGroupZIDFromSecurityGroupRulesZID(SecurityGroupZID) + "-sgrs-id"
```

Comment:
After few discussions, let's not add suffix for this resource. It'll create confusion for users who will import this resource, or are already familiarised with other resources.